docs: docs folder for new asyncapi community section (#442)
asyncapi/community: docs/how-to-create-new-repository.md

> # Create new AsyncAPI repository

Reviewer: ```suggestion

```

lets leave below intro without heading

Author: Okay, if I understand you correctly we are using your suggestion as the name of the file, replacing the primary name.

If yes, hope that is fine as I have already committed your suggestions.

> This guide explains the _approval process_ and the _default repository configurations_ required for creating a new repository in the [AsyncAPI Initiative GitHub organization](https://github.com/asyncapi). We also cover how to _donate a project_ to the AsyncAPI Initiative by transferring repository ownership and becoming its maintainer. 

Reviewer: when I looked into PR https://github.com/asyncapi/community/pull/449 and this one, I figured we mix content too much. 

I suggest that this document focuses only on what technical steps are taken by org admins to create repo, and first part of this document, the approval process, pull into separate markdown file please and call it `approval-process-new-or-donated-project.md`

I hope it makes sense

Author: To be honest, it doesn't make sense to me but I can't explain why maybe the name of the file is just too long, and thinking that there is not going to be much content for this process.

Not sure if it is necessary to separate it.

Reviewer: the problem is that the current document is a mix of content for different audience:
- `Getting a new AsyncAPI repository approved` should be a separate document, as it is for the community members that plan to donate a project or start new one at AsyncAPI. It describes process
- other sections are nothing else but technical instructions for people that create repository and configure it after the approval process is completed

2 different groups of people.

Makes sense?

Author: Yes, it does, I will have to move some parts of the document into `Getting a new AsyncAPI repository approved`.

Author: > when I looked into PR #449 and this one, I figured we mix content too much.
> 
> I suggest that this document focuses only on what technical steps are taken by org admins to create repo, and first part of this document, the approval process, pull into separate markdown file please and call it `approval-process-new-or-donated-project.md`
> 
> I hope it makes sense

I created a separate file called "Donate-project.md" and a file called "Getting a new AsyncAPI repository approved.md"

While I changed "how to create a new repository.md"  to "Create Repository as an admin.md"

With the understanding that the docs "Create repository as an admin" is a technical step taken by org admins to create a repo, those technical steps involve Github action, Configuration, and sonar cloud, right?

> 

Reviewer: ```suggestion

## Getting a new AsyncAPI repository approved
```

> Let's create a new AsyncAPI repository:

Reviewer: ```suggestion
In case you want to donate existing project to AsyncAPI organization, or start a new project, follow below steps:
```

6. Invite new maintainers as repo admins.

## GitHub Actions: CI/CD workflow
Here are the steps `@derberg` or `@fmvilas` take when running AsyncAPI `CI/CD` workflows:
1. Go to Asyncapi `/.github` repo.
2. Click `Action` to view all workflows.
3. Select `Global workflow to rule them all`.
4. Click `run workflow` and paste the name of the new repo. Running the workflow takes all the workflows from the AsyncAPI `.github` repo and runs those on the specified repo.

The AsyncAPI bot creates a set of `pull requests` for a given repo. You need to `merge` them as a repo's maintainer.

<Remember>
 Manage project automation independently if you don't want to use the AsyncAPI global workflow.
</Remember>
<Remember>
Repos that shouldn't get CI/CD updates must do the following:
Open a `pull request` in the `.github` repo. 
Edit the`global_replicator.yml` in the `.github` repo; add the newly created repo to every `Job` under **repos to ignore**. (Example: `repos_to_ignore: your-repo-name`) 
</Remember>

## Configure repository

> All AsyncAPI repositories should be similar in structure, settings, and restrictions. 

Reviewer: ```suggestion
All AsyncAPI repositories start with the same settings and restrictions. It is up to maintainers of given repository to modify settings if these disrupt maintenance. 
```

Make sure the **Automatically delete head branches** option is enabled.

## SonarCloud scans
Integrate each AsyncAPI repository with `https://sonarcloud.io/organizations/asyncapi/projects` for automated quality and security scans. 

Add your project to the list via: `https://sonarcloud.io/projects/create`.

<Remember>
Our `CI/CD workflows` fail if your repository has a `main branch` because we currently support only `master branch` in our projects.

Should you accidentally forget and create a repository with a `main branch`, remember to rename it to `master branch`. 
</Remember>
 
## Branch protection rules
Branch protection must be added by whoever creates the repository. 

Let's walk through the steps:
1. Under the **Setting** tab, click `branches`.
2. Change the branch name to `master`.
3. **Enable** all of the following settings under **Protect matching branch**:
- `Require a pull request before merging`
- `Request approvals`
- `Dismiss stale pull request approvals when new commits are pushed`
- `Require review from Code Owners`
- `Restrict who can dismiss pull request reviews` _(admin only)_

> - `Require status check before merging` _(This status check runs Github action workflows before merging pull requests.)_

Reviewer: ```suggestion
- `Require status check before merging` _(This status check runs Github action workflows before merging pull requests.)_ and select `Lint PR title` and pull request testing related workflows
```

- `Require branch to be up to date before merging`
- `Do not allow bypassing the above settings`
- `Restrict who can push to matching branches`

> - `Restrict pushes that create matching branches`

Reviewer: ```suggestion

```

this one is not needed

## Collaborators and teams
The AsyncAPI Initiative grants admin roles to project maintainers.

<Remember>
Don't forget to add the `AsyncAPI bot` as a maintainer to each project.
</Remember>

## Codeowners file
The Codeowners file contains the code owners' names; this includes both the admins and the `asyncapi-bot-eve`.

Checkout the following [`Codeowners file` example from the `/website` repository](https://github.com/asyncapi/website/blob/master/CODEOWNERS):
```
# This file provides an overview of code owners in this repository.
	 
	# Each line is a file pattern followed by one or more owners.
	# The last matching pattern has the most precedence.
	# For more details, read the following article on GitHub: https://help.github.com/articles/about-codeowners/.
	 
	# The default owners are automatically added as reviewers when you open a pull request unless different owners are specified in the file.
	* @fmvilas @derberg @mcturco @akshatnema @magicmatatjahu @asyncapi-bot-eve
	 
	# All .md files
	*.md @alequetzalli @asyncapi-bot-eve
	 
	README.md @alequetzalli @fmvilas @derberg @mcturco @akshatnema @magicmatatjahu @asyncapi-bot-eve
```

> ## Donate project to AsyncAPI

Reviewer: this section is like negation to the whole document 😅  Previous sections are about donation process, getting stuff approved and later configured. And here all suddenly we say something different.

RabbitMQ mention is useful, but not in this context.
https://github.com/orgs/asyncapi/discussions/223 is basically an example of `Create new AsyncAPI repository` in action, so maybe just add it at the end of steps, as a note, and link. Example:

```
> In case you want to see an example process of project donation. Check out [discussion about donating RabbitMQ template](https://github.com/orgs/asyncapi/discussions/223)

```

Author: I don't understand this comment

Reviewer: @Dule-martins it is not helping if you just say that you do not understand the comment. It was a long comment, and to help you out, I need more specific information about what is not clear.

Please have a look at the content you provided in this `Donate project to AsyncAPI` section. What does it say? and what does the rest of the document say? totally different things 🤷🏼

Author: Oh, my bad! So sorry about that.
I will definitely do better. 
Thank you, alright let me check.

6. Invite new maintainers as repo admins.

## GitHub Actions: CI/CD workflow
Here are the steps `@derberg` or `@fmvilas` take when running AsyncAPI `CI/CD` workflows:
1. Go to Asyncapi `/.github` repo.
2. Click `Action` to view all workflows.
3. Select `Global workflow to rule them all`.
4. Click `run workflow` and paste the name of the new repo. Running the workflow takes all the workflows from the AsyncAPI `.github` repo and runs those on the specified repo.

The AsyncAPI bot creates a set of `pull requests` for a given repo. You need to `merge` them as a repo's maintainer.

<Remember>
 Manage project automation independently if you don't want to use the AsyncAPI global workflow.
</Remember>
<Remember>
Repos that shouldn't get CI/CD updates must do the following:
Open a `pull request` in the `.github` repo. 
Edit the`global_replicator.yml` in the `.github` repo; add the newly created repo to every `Job` under **repos to ignore**. (Example: `repos_to_ignore: your-repo-name`) 
</Remember>

## Configure repository

> All AsyncAPI repositories should be similar in structure, settings, and restrictions. 

Author: I feel this is not complete, if you are saying all asyncapi repositories should be similar in structure, settings, and restrictions then there should be a step to how I can achieve that as a user of the documentation.

What if is possible to have the steps laid out?

Reviewer: I suggested change for this sentence, how about now?

Author: Okay, that's fine I will suggest this `All AsyncAPI repository shares similar structure, settings, and restrictions.`

# How to create new repository

> This is  not about creating a repository on Github, so you will not be reading about that, but you can check [here](https://docs.github.com/en/get-started/quickstart/create-a-repo)

Reviewer: ```suggestion
This document is not about creating a repository on GitHub. If this is what you were looking for, then [read GitHub guides on repository creation](https://docs.github.com/en/get-started/quickstart/create-a-repo).
```

it is important to have descriptive links for better accessibility for people using screen readers

> This docs is about creating a new project in [asyncapi](https://www.asyncapi.com/), it is the guide to how you can create a repository in [asyncapi organisation on Github](https://github.com/asyncapi), who you need to talk, and how you should configure the repository.

Reviewer: ```suggestion
This document is about creating a new repository at [AsyncAPI Initiative GitHub organization](https://github.com/asyncapi). This guide explains what is the approval process and default repository configuration options.
```

> This docs is also helpful if you are donating a project to [asyncapi](https://www.asyncapi.com/), which involves transfering a repository into [asyncapi organization](https://github.com/asyncapi) and become an onwer of that project.

Reviewer: ```suggestion
This document is also helpful if you are donating a project to AsyncAPI Initiative, which involves transferring repository ownership, but still becoming its maintainer.
```

> -----------------------------

Reviewer: ```suggestion

```

we do not use these lines in docs

> ## Creating a Repo for new project in Asyncapi

Reviewer: ```suggestion
## Creating a repo for a new project
```

> 1. Visit [asyncapi initaitive on Github](https://github.com/asyncapi) click on `Discussion tab` to start new discussion.

Reviewer: ```suggestion
1. Visit [AsyncAPI Initiative Discussions](https://github.com/orgs/asyncapi/discussions)
```

less clicking with direct link 😉

> ![alt text](../assets/image-for-docs/discussion.PNG "New Disccusion on Github")

Reviewer: we can remove both screen shots, they are not adding much to the document, and also `New discussion` button is pretty much visible in the UI

Reviewer 2: Ok, so I should take off the screenshot and the part that talks about the new discussion tab

> 2. Click on `New discussion`, select `Category` add a `title`, ping [Lukasz Gornick](https://github.com/derberg) or [Fran Méndez](https://github.com/fmvilas) in your discussion about the new project.

Reviewer: ```suggestion
2. Start a new discussion by clicking on `New discussion` and selecting the `Idea` category.

    Specify in the title that you want to create a new repository. In the description specify what is the purpose of the new repository, and why it should be created under the AsyncAPI organization. See also [example discussion from the past](https://github.com/orgs/asyncapi/discussions/300).
```

> ![alt text](../assets/image-for-docs/Github-discussion.jpg "Start a new discussion")

Reviewer: please remove the screen shot

> ### Note: 

Reviewer: please remove the note, better if folks use one category

Reviewer 2: Okay, I will

There are 9 categories to every dicussion that been discuss here on asyncapi organization, each of these ctategories represent what you are proposing to the technical steering commitee.

> 3. After starting a discussion about your new project for asyncapi, and successfully convienced other members of the community why your project ideas is necessary and you want to maintain the codebase.

Reviewer: ```suggestion
3. Additionally specify the initial maintainers of the project. At least 2 are needed to start

4. Your proposal for a new repository must be approved by Technical Steering Committee (TSC). So make sure that at the end of proposal description you mention @asyncapi/tsc_members GitHub team to get TSC members notified.
```

# How to create a new repository

This document is not about creating a repository on GitHub. If this is what you were looking for, then [read GitHub's guides on repository creation](https://docs.github.com/en/get-started/quickstart/create-a-repo).

This document is about creating a new repository in the [AsyncAPI Initiative GitHub organization](https://github.com/asyncapi). This guide explains what is the approval process and default repository configuration options.

This document is also helpful if you donate a project to the AsyncAPI Initiative, which involves transferring repository ownership and becoming its maintainer.

## Creating a repo for a new project

1. Visit [AsyncAPI Initiative Discussions](https://github.com/orgs/asyncapi/discussions).

2. Start a new discussion by clicking on `New discussion` and selecting the `Idea` category. Specify in the title that you want to create a new repository. In the description specify what is the purpose of the new repository, and why it should be created under the AsyncAPI organization. See also [example discussion from the past](https://github.com/orgs/asyncapi/discussions/300).

> 3. After starting a discussion about your new project for AsyncAPI and successfully convinced other community members why your project ideas are necessary, and you want to maintain the codebase.

Reviewer 2: What are you suggesting?

> 3. After starting a discussion about your new project for AsyncAPI and successfully convinced other community members why your project ideas are necessary, and you want to maintain the codebase.

Reviewer: I do not know what to suggest as I do not know what is your message here

> 3. After starting a discussion about your new project for AsyncAPI and successfully convinced other community members why your project ideas are necessary, and you want to maintain the codebase.

Reviewer 2: I tried out something else kindly take a look

> 4. Additionally, when creating a proposal, specify the initial maintainers of the project and whether you want to use AsyncAPI CI/CD workflows or not, so we don't need to ask you. At least 2 code maintainers are needed to start.

Reviewer 2: step 4 comes before step 3, ok.

> 5. Your proposal for a new repository must be approved by the Technical Steering Committee (TSC). Make sure that at the end of the proposal description you mention the `@asyncapi/tsc_members` GitHub team so as notify all TSC members.

Reviewer 2: Yes, I corrected that in the next PR

To learn more, watch this [AsyncAPI CI/CD video](https://www.youtube.com/watch?v=DsQfmlc3Ubo)

## Github Actions CI/CD in new repo
Using AsyncAPI `CI/CD` workflow saves you from a lot of work load, after including it in your Github discussion/proposal specifing that you want to use our `CI/CD` workflow, the following activities will be carried out by Lukasz or Fran.

1. Go to Asyncapi `/.github` repo.
2. Click on `Action` to view all workflows.
3. Select `Global workflow to rule then all`.
4. Click on `run workflow` and paste the name of the new repo.
5. `Run workflow`.

We have a `workflow` that takes all the workflows from `.github` and allows you to push to a specific repo selectively. One of the owners (which at the time of this writing is still Lukasz or Fran) executes these actions.

The AsyncAPI bot creates a set of `pull requests` for a given repo. You need to `merge` them as a repo's maintainer.

If you don't want to use the AsyncAPI global workflow, you will have to automate all the necessary `pull requests` that the bot would have made.

> We open a `pull request` in the `.github` repo, select `global_replicator.yml` and add the newly created repo to every single `Job` from the `global_replicator.yml` file. Add as **list of repos to be ignored** in `repos_to_ignore: name-of-repo`

Reviewer: don't write We but rather, that "what user have to do, to make sure their repo never gets CI/CD updates"
